identity/app/actions/AuthenticatedActions.scala

```diff
@@ -87,6 +95,22 @@ class AuthenticatedActions(
       }
   }
 
+  def apiUserShouldRepermissionRefiner: ActionRefiner[AuthRequest, AuthRequest] = new ActionRefiner[AuthRequest, AuthRequest] {
+    override val executionContext = ec
+    //TODO: verify if the user reperm'd instead of validate. also verify v1 email subs
+    def refine[A](request: AuthRequest[A]) = Future.successful {
+      if(IdentityRedirectUsersWithLingeringV1ConsentsSwitch.isSwitchedOn) {
+        request.user.statusFields.userEmailValidated match {
```

Comment:
I would leave this until we actually set a hasRepermissioned flag to the User model?